### ISSUE TITLE
getGroupMembershipForUser domain syntax

### ISSUE THREAD

**Issue by bjorke** [gheeres/node-activedirectory#143]
When doing an authenticate call, 'domain\\username' is a valid syntax
```
var username = 'domain\\username';
var password = 'password'; 
ad.authenticate(username, password, function(err, auth) 
```

 but when doing it with the getGroupMembershipForUser it won't find the user
```
var sAMAccountName = 'domain\\username';
adGroups.getGroupMembershipForUser(sAMAccountName, function(err, groups) 
```

> User: domain\user not found.

The real issue here is that the `@domain` works for authenticating the user but don't work with the getGroupMembershipForUser.
I'm assuming it's due to the domain name not being the same as the userPrincipalName.

It seams to be working if I don't specify any domain at all with the getGroupMembershipForUser call, but I'm not sure I can guarantee that it's a unique username then.

Other than that it works like a charm!

**Comment by de1m** [comment]
Yes, I've the same error, workaround is to use only the username for this function.
But it's working with some users (like: user@domain.local).

**Comment by bjorke** [comment]
The main problem with that is since we've got several domains the same username can at least in theory occur twice.
Not an issue on our end since that would cause problems in other applications, but it's still a potential problem and it's going to cause problems for someone.
